lib/block_view_validator.go

```diff
@@ -1571,6 +1583,13 @@ func (bav *UtxoView) _disconnectUnjailValidator(
 	}
 	bav._setValidatorEntryMappings(prevValidatorEntry)
 
+	// Restore the PrevGlobalActiveStakeAmountNanos.
+	prevGlobalActiveStakeAmountNanos := operationData.PrevGlobalActiveStakeAmountNanos
+	if prevGlobalActiveStakeAmountNanos == nil {
+		return errors.New("_disconnectUnjailValidator: PrevGlobalActiveStakeAmountNanos is nil")
```

Comment:
Nit: Whenever something should never happen, like I think is the case here, I like to add "This should never happen" or "sanity-check failed" to indicate that in the logs for whoever ends up having to debug it.

Reply:
Let's add this^